### PR TITLE
WIP: use chrome-launcher for handling the browser.

### PR DESCRIPTION
**Pull request by paulirish**
this is the launcher module from lighthouse that we maintain.
https://www.npmjs.com/package/chrome-launcher

it defaults to an open port

addresses #21

### CODE REVIEW

File: lib/Browser.js [hunk 1]
```diff
@@ -98,15 +94,17 @@ class Browser {
...
         // Cleanup as processes exit.
         process.on('exit', () => this._chromeProcess.kill());
-        this._chromeProcess.on('exit', () => {
-            this._terminated = true;
-            removeRecursive(this._userDataDir);
-        });
-
-        await waitForChromeResponsive(this._remoteDebuggingPort);
+        // this._chromeProcess.on('exit', () => {
+        //     this._terminated = true;
```
Comment (aslushnikov) [inline lib/Browser.js]: can we have this back? I think it's still valuable

File: package.json [hunk 2]
```diff
@@ -16,6 +16,7 @@
   "author": "The Chromium Authors",
   "license": "SEE LICENSE IN LICENSE",
   "dependencies": {
+    "chrome-launcher": "^0.1.3",
```
Comment (aslushnikov) [inline package.json:19]: Does chrome-launcher depend on lighthouse? (i see it [here](https://github.com/GoogleChrome/lighthouse/blob/master/chrome-launcher/package.json#L22)) This seems to be a pretty heavy dependency to pull in, is it really needed?

File: lib/Browser.js [hunk 3]
```diff
@@ -36,15 +38,9 @@ class Browser {
      */
     constructor(options) {
         options = options || {};
-        ++browserId;
```
Comment (aslushnikov) [inline lib/Browser.js]: let's also kill it from L26

**Review by aslushnikov — COMMENTED** [review]
That's exciting!

File: package.json (hunk 2 above)
Comment (ebidel) [reply package.json:19]: [PR](https://github.com/GoogleChrome/lighthouse/pull/2528) landing very soon to fix that.

Comment (paulirish) [reply package.json:19]: `chrome-launcher@0.2.0` shipped without LH as a dep.

Comment (paulirish) [reply package.json:19]: fixed.

**Comment by paulirish** [comment]
Updated and rebased.

However we still need `launchedChrome.on('exit', fn)` support from chrome-launcher to complete this.

**Comment by paulirish** [comment]
Closing until this is ready.

**Comment by ghost** [comment]
Hi, can we use chrome-launcher with puppeteer now?

**Comment by ghost** [comment]
I copy some code from chrome-launcher to make it work, but it looks so dirty. Is there a better way to archive that? Thanks. 😉 

```javascript
const { getPlatform } = require('chrome-launcher/utils')
const chromeFinder = require('chrome-launcher/chrome-finder')
const Puppeteer = require('puppeteer')

async function setOptionsDefaults(options)
{
  if (options.executablePath === undefined) {
    const installations = await chromeFinder[getPlatform()]()
    if (installations.length === 0) {
      throw new Error('No Chrome Installations Found')
    }
    options.executablePath = installations[0]
  }

  return options
}

class Puppeteer2 extends Puppeteer {
  static async launch(options) {
    options = await setOptionsDefaults(options)
    return super.launch(options)
  }
}

module.exports = Puppeteer2
```

**Pull request closed without merging**